### DISCUSSION TITLE
assertion

### DISCUSSION THREAD

**Discussion by behrj** [seqan/seqan3#3110] [Q&A]
Hey there, 

Not sure this is the right place to write this, but I also could not find another one. If you have a forum somewhere that you are using more actively, please let me know. 
I started switching to seqan3 recently and I ran into one thing that I cannot easily resolve by myself. Doing a banded alignment on a number of rather small sequences runs into this assertion below deterministically. I looked precisely at the pair of sequences that trigger the assertion, but they look fine and without running the whole batch before, I cannot trigger the issue. 
When I remove the config for banded alignment, then everything is smooth. 

If you have an idea, how I could reproduce that with a minimal dataset, that I could also share, please let me know. 

Best, 
Jonas

seqan3/include/seqan3/alignment/decorator/gap_decorator.hpp:683: void seqan3::gap_decorator<inner_type>::basic_iterator<<anonymous> >::jump(seqan3::gap_decorator<inner_type>::size_type) [with bool <anonymous> = true; inner_type = std::span<seqan3::dna5, 18446744073709551615>; seqan3::gap_decorator<inner_type>::size_type = long unsigned int]: Assertion `new_pos <= host->size()' failed.

**Comment by marehr** [comment]
Hi @behrj! 

If I needed to debug it, I would either print out the sequences to the terminal and re-try the application with just those sequences (i.e. making the set smaller and smaller by commenting them out), or use for example [Visual Studio Code](https://code.visualstudio.com/) and the built-in debugger to see the trace and the state that your application is in, and why it behaves differently with a set vs. two-sequences.

As you get assertion errors, I assume that you build with `cmake -DCMAKE_BUILD_TYPE=Debug <..>`.
Can you rerun your batch with `valgrind ./myapp --param1 --param2`, maybe this will give us some hints.
This can take > 10x the normal application runtime, but will give you a complete stack trace with functions that are involved.

Your stated assertion 

https://github.com/seqan/seqan3/blob/cdff3ddfd26367c0fe7bdc1bacbbcbfd30e9264c/include/seqan3/alignment/decorator/gap_decorator.hpp#L680-L683

does hint that somewhere in the code, a character of an aligned sequence (= sequence + gaps, e.g. "AC-GT-AC-T") that is represented by the `gap_decorator` will be accessed out-of-range. It would be interesting if your application has faulty logic, or seqan internally messes something up. Maybe we can figure that out if you provide the stack trace.


BTW I think you could also have opened an issue, but I guess here is fine too.

**Comment by behrj** [comment]
Hi Marcel, thanks for the quick response. 
I had checked with valgrind because I was assuming there is an memory error on my side, but that does not seem to be the case. Please find the stack trace below, I hope this is going to tell you something. 
And thanks for looking into this. 
I will work on trying to reduce the set of sequences to reproduce the problem. 

Cheers,
Jonas 

detect_fusions_banded: seqan3/include/seqan3/alignment/decorator/gap_decorator.hpp:683: void seqan3::gap_decorator<inner_type>::basic_iterator<<anonymous> >::jump(seqan3::gap_decorator<inner_type>::size_type) [with bool <anonymous> = true; inner_type = std::span<seqan3::dna5, 18446744073709551615>; seqan3::gap_decorator<inner_type>::size_type = long unsigned int]: Assertion `new_pos <= host->size()' failed.
==47==
==47== Process terminating with default action of signal 6 (SIGABRT)
==47==    at 0x4DA10C0: __pthread_kill_implementation (pthread_kill.c:44)
==47==    by 0x4D5AB2B: raise (raise.c:26)
==47==    by 0x4D474BB: abort (abort.c:79)
==47==    by 0x4D54433: __assert_fail_base (assert.c:92)
==47==    by 0x4D5449B: __assert_fail (assert.c:101)
==47==    by 0x178C9B: seqan3::gap_decorator<std::span<seqan3::dna5, 18446744073709551615ul> >::basic_iterator<true>::jump(unsigned long) (gap_decorator.hpp:683)

https://github.com/seqan/seqan3/blob/fe7458e83b1415db38ff8148597017c3fabd030a/include/seqan3/alignment/decorator/gap_decorator.hpp#L681-L684

==47==    by 0x1AD56B: operator+= (gap_decorator.hpp:799)

https://github.com/seqan/seqan3/blob/fe7458e83b1415db38ff8148597017c3fabd030a/include/seqan3/alignment/decorator/gap_decorator.hpp#L797-L801

==47==    by 0x1AD56B: fill_aligned_sequence<std::ranges::reverse_view<std::ranges::ref_view<std::vector<std::pair<seqan3::detail::trace_directions, long unsigned int> > > >, seqan3::gap_decorator<std::span<seqan3::dna5, 18446744073709551615> >, seqan3::gap_decorator<std::span<seqan3::dna5, 18446744073709551615> > > (aligned_sequence_builder.hpp:243)

https://github.com/seqan/seqan3/blob/fe7458e83b1415db38ff8148597017c3fabd030a/include/seqan3/alignment/matrix/detail/aligned_sequence_builder.hpp#L221-L245

<details><summary>==47==    by 0x1AD56B: seqan3::detail::aligned_sequence_builder</summary>
==47==    by 0x1AD56B: seqan3::detail::aligned_sequence_builder<std::span<seqan3::dna5, 18446744073709551615ul>&, std::span<seqan3::dna5, 18446744073709551615ul>&>::result_type seqan3::detail::aligned_sequence_builder<std::span<seqan3::dna5, 18446744073709551615ul>&, std::span<seqan3::dna5, 18446744073709551615ul>&>::operator()<std::ranges::subrange<seqan3::detail::trace_iterator_banded<seqan3::detail::two_dimensional_matrix<seqan3::detail::trace_directions, std::allocator<seqan3::detail::trace_directions>, (seqan3::detail::matrix_major_order)0>::basic_iterator<false> >, std::default_sentinel_t, (std::ranges::subrange_kind)0> >(std::ranges::subrange<seqan3::detail::trace_iterator_banded<seqan3::detail::two_dimensional_matrix<seqan3::detail::trace_directions, std::allocator<seqan3::detail::trace_directions>, (seqan3::detail::matrix_major_order)0>::basic_iterator<false> >, std::default_sentinel_t, (std::ranges::subrange_kind)0>&&) (aligned_sequence_builder.hpp:207)
</details>

https://github.com/seqan/seqan3/blob/fe7458e83b1415db38ff8148597017c3fabd030a/include/seqan3/alignment/matrix/detail/aligned_sequence_builder.hpp#L171-L212

<details><summary>==47==    by 0x1AD7CF: void seqan3::detail::alignment_algorithm<...></summary>
==47==    by 0x1AD7CF: void seqan3::detail::alignment_algorithm<seqan3::configuration<seqan3::align_cfg::method_global, seqan3::align_cfg::scoring_scheme<seqan3::nucleotide_scoring_scheme<signed char> >, seqan3::align_cfg::gap_cost_affine, seqan3::align_cfg::output_score, seqan3::align_cfg::output_begin_position, seqan3::align_cfg::output_end_position, seqan3::align_cfg::band_fixed_size, seqan3::align_cfg::detail::result_type<seqan3::alignment_result<seqan3::detail::alignment_result_value_type<std::nullopt_t*, std::nullopt_t*, int, seqan3::detail::advanceable_alignment_coordinate<(seqan3::detail::advanceable_alignment_coordinate_state)0>, seqan3::detail::advanceable_alignment_coordinate<(seqan3::detail::advanceable_alignment_coordinate_state)0>, std::nullopt_t*, std::nullopt_t*, std::nullopt_t*> > > >, seqan3::detail::deferred_crtp_base<seqan3::detail::alignment_matrix_policy, seqan3::detail::alignment_score_matrix_one_column_banded<int>, seqan3::detail::alignment_trace_matrix_full_banded<seqan3::detail::trace_directions, false> >, seqan3::detail::deferred_crtp_base<seqan3::detail::affine_gap_policy, int, std::integral_constant<bool, false> >, seqan3::detail::deferred_crtp_base<seqan3::detail::find_optimum_policy>, seqan3::detail::deferred_crtp_base<seqan3::detail::affine_gap_init_policy>, seqan3::detail::deferred_crtp_base<seqan3::detail::scoring_scheme_policy, seqan3::nucleotide_scoring_scheme<signed char> > >::make_alignment_result<unsigned long, std::span<seqan3::dna5, 18446744073709551615ul>, std::span<seqan3::dna5, 18446744073709551615ul>, std::function<void (seqan3::alignment_result<seqan3::detail::alignment_result_value_type<std::nullopt_t*, std::nullopt_t*, int, seqan3::detail::advanceable_alignment_coordinate<(seqan3::detail::advanceable_alignment_coordinate_state)0>, seqan3::detail::advanceable_alignment_coordinate<(seqan3::detail::advanceable_alignment_coordinate_state)0>, std::nullopt_t*, std::nullopt_t*, std::nullopt_t*> >)> >(unsigned long, std::span<seqan3::dna5, 18446744073709551615ul>&, std::span<seqan3::dna5, 18446744073709551615ul>&, std::function<void (seqan3::alignment_result<seqan3::detail::alignment_result_value_type<std::nullopt_t*, std::nullopt_t*, int, seqan3::detail::advanceable_alignment_coordinate<(seqan3::detail::advanceable_alignment_coordinate_state)0>, seqan3::detail::advanceable_alignment_coordinate<(seqan3::detail::advanceable_alignment_coordinate_state)0>, std::nullopt_t*, std::nullopt_t*, std::nullopt_t*> >)>&) [clone .isra.0] (alignment_algorithm.hpp:636)
==47==    by 0x1ADA1F: compute_single_pair<std::vector<seqan3::dna5>&, std::vector<seqan3::dna5>&, std::function<void(seqan3::alignment_result<seqan3::detail::alignment_result_value_type<std::nullopt_t*, std::nullopt_t*, int, seqan3::detail::advanceable_alignment_coordinate<>, seqan3::detail::advanceable_alignment_coordinate<>, std::nullopt_t*, std::nullopt_t*, std::nullopt_t*> >)> > (alignment_algorithm.hpp:304)
</details>

https://github.com/seqan/seqan3/blob/fe7458e83b1415db38ff8148597017c3fabd030a/include/seqan3/alignment/pairwise/alignment_algorithm.hpp#L283-L311

==47==    by 0x1ADA1F: operator()<std::ranges::subrange<seqan3::detail::zip_view<std::ranges::single_view<std::tuple<std::vector<seqan3::dna5, std::allocator<seqan3::dna5> >&, std::vector<seqan3::dna5, std::allocator<seqan3::dna5> >&> >, std::ranges::iota_view<int, std::unreachable_sentinel_t> >::iterator<false>, seqan3::detail::zip_view<std::ranges::single_view<std::tuple<std::vector<seqan3::dna5, std::allocator<seqan3::dna5> >&, std::vector<seqan3::dna5, std::allocator<seqan3::dna5> >&> >, std::ranges::iota_view<int, std::unreachable_sentinel_t> >::iterator<false>, (std::ranges::subrange_kind)1>, std::function<void(seqan3::alignment_result<seqan3::detail::alignment_result_value_type<std::nullopt_t*, std::nullopt_t*, int, seqan3::detail::advanceable_alignment_coordinate<>, seqan3::detail::advanceable_alignment_coordinate<>, std::nullopt_t*, std::nullopt_t*, std::nullopt_t*> >)> > (alignment_algorithm.hpp:200)

https://github.com/seqan/seqan3/blob/fe7458e83b1415db38ff8148597017c3fabd030a/include/seqan3/alignment/pairwise/alignment_algorithm.hpp#L193-L201

<details><summary>==47==    by 0x1ADA1F: __invoke_impl</summary>

==47==    by 0x1ADA1F: __invoke_impl<void, seqan3::detail::alignment_algorithm<seqan3::configuration<seqan3::align_cfg::method_global, seqan3::align_cfg::scoring_scheme<seqan3::nucleotide_scoring_scheme<signed char> >, seqan3::align_cfg::gap_cost_affine, seqan3::align_cfg::output_score, seqan3::align_cfg::output_begin_position, seqan3::align_cfg::output_end_position, seqan3::align_cfg::band_fixed_size, seqan3::align_cfg::detail::result_type<seqan3::alignment_result<seqan3::detail::alignment_result_value_type<std::nullopt_t*, std::nullopt_t*, int, seqan3::detail::advanceable_alignment_coordinate<(seqan3::detail::advanceable_alignment_coordinate_state)0>, seqan3::detail::advanceable_alignment_coordinate<(seqan3::detail::advanceable_alignment_coordinate_state)0>, std::nullopt_t*, std::nullopt_t*, std::nullopt_t*> > > >, seqan3::detail::deferred_crtp_base<seqan3::detail::alignment_matrix_policy, seqan3::detail::alignment_score_matrix_one_column_banded<int>, seqan3::detail::alignment_trace_matrix_full_banded<seqan3::detail::trace_directions, false> >, seqan3::detail::deferred_crtp_base<seqan3::detail::affine_gap_policy, int, std::integral_constant<bool, false> >, seqan3::detail::deferred_crtp_base<seqan3::detail::find_optimum_policy>, seqan3::detail::deferred_crtp_base<seqan3::detail::affine_gap_init_policy>, seqan3::detail::deferred_crtp_base<seqan3::detail::scoring_scheme_policy, seqan3::nucleotide_scoring_scheme<signed char> > >&, std::ranges::subrange<seqan3::detail::zip_view<std::ranges::single_view<std::tuple<std::vector<seqan3::dna5, std::allocator<seqan3::dna5> >&, std::vector<seqan3::dna5, std::allocator<seqan3::dna5> >&> >, std::ranges::iota_view<int, std::unreachable_sentinel_t> >::iterator<false>, seqan3::detail::zip_view<std::ranges::single_view<std::tuple<std::vector<seqan3::dna5, std::allocator<seqan3::dna5> >&, std::vector<seqan3::dna5, std::allocator<seqan3::dna5> >&> >, std::ranges::iota_view<int, std::unreachable_sentinel_t> >::iterator<false>, (std::ranges::subrange_kind)1>, std::function<void(seqan3::alignment_result<seqan3::detail::alignment_result_value_type<std::nullopt_t*, std::nullopt_t*, int, seqan3::detail::advanceable_alignment_coordinate<(seqan3::detail::advanceable_alignment_coordinate_state)0>, seqan3::detail::advanceable_alignment_coordinate<(seqan3::detail::advanceable_alignment_coordinate_state)0>, std::nullopt_t*, std::nullopt_t*, std::nullopt_t*> >)> > (invoke.h:61)
==47==    by 0x1ADA1F: __invoke_r<void, seqan3::detail::alignment_algorithm<seqan3::configuration<seqan3::align_cfg::method_global, seqan3::align_cfg::scoring_scheme<seqan3::nucleotide_scoring_scheme<signed char> >, seqan3::align_cfg::gap_cost_affine, seqan3::align_cfg::output_score, seqan3::align_cfg::output_begin_position, seqan3::align_cfg::output_end_position, seqan3::align_cfg::band_fixed_size, seqan3::align_cfg::detail::result_type<seqan3::alignment_result<seqan3::detail::alignment_result_value_type<std::nullopt_t*, std::nullopt_t*, int, seqan3::detail::advanceable_alignment_coordinate<(seqan3::detail::advanceable_alignment_coordinate_state)0>, seqan3::detail::advanceable_alignment_coordinate<(seqan3::detail::advanceable_alignment_coordinate_state)0>, std::nullopt_t*, std::nullopt_t*, std::nullopt_t*> > > >, seqan3::detail::deferred_crtp_base<seqan3::detail::alignment_matrix_policy, seqan3::detail::alignment_score_matrix_one_column_banded<int>, seqan3::detail::alignment_trace_matrix_full_banded<seqan3::detail::trace_directions, false> >, seqan3::detail::deferred_crtp_base<seqan3::detail::affine_gap_policy, int, std::integral_constant<bool, false> >, seqan3::detail::deferred_crtp_base<seqan3::detail::find_optimum_policy>, seqan3::detail::deferred_crtp_base<seqan3::detail::affine_gap_init_policy>, seqan3::detail::deferred_crtp_base<seqan3::detail::scoring_scheme_policy, seqan3::nucleotide_scoring_scheme<signed char> > >&, std::ranges::subrange<seqan3::detail::zip_view<std::ranges::single_view<std::tuple<std::vector<seqan3::dna5, std::allocator<seqan3::dna5> >&, std::vector<seqan3::dna5, std::allocator<seqan3::dna5> >&> >, std::ranges::iota_view<int, std::unreachable_sentinel_t> >::iterator<false>, seqan3::detail::zip_view<std::ranges::single_view<std::tuple<std::vector<seqan3::dna5, std::allocator<seqan3::dna5> >&, std::vector<seqan3::dna5, std::allocator<seqan3::dna5> >&> >, std::ranges::iota_view<int, std::unreachable_sentinel_t> >::iterator<false>, (std::ranges::subrange_kind)1>, std::function<void(seqan3::alignment_result<seqan3::detail::alignment_result_value_type<std::nullopt_t*, std::nullopt_t*, int, seqan3::detail::advanceable_alignment_coordinate<(seqan3::detail::advanceable_alignment_coordinate_state)0>, seqan3::detail::advanceable_alignment_coordinate<(seqan3::detail::advanceable_alignment_coordinate_state)0>, std::nullopt_t*, std::nullopt_t*, std::nullopt_t*> >)> > (invoke.h:111)
==47==    by 0x1ADA1F: _ZNSt17_Function_handlerIFvNSt6ranges8subrangeIN6seqan36detail8zip_viewIJNS0_11single_viewISt5tupleIJRSt6vectorINS2_4dna5ESaIS8_EESB_EEEENS0_9iota_viewIiSt22unreachable_sentinel_tEEEE8iteratorILb0EEESJ_LNS0_13subrange_kindE1EEESt8functionIFvNS2_16alignment_resultINS3_27alignment_result_value_typeIPSt9nullopt_tSQ_iNS3_32advanceable_alignment_coordinateILNS3_38advanceable_alignment_coordinate_stateE0EEEST_SQ_SQ_SQ_EEEEEEENS3_19alignment_algorithmINS2_13configurationIJNS2_9align_cfg13method_globalENS11_14scoring_schemeINS2_25nucleotide_scoring_schemeIaEEEENS11_15gap_cost_affineENS11_12output_scoreENS11_21output_begin_positionENS11_19output_end_positionENS11_15band_fixed_sizeENS11_6detail11result_typeISV_EEEEEJNS3_18deferred_crtp_baseINS3_23alignment_matrix_policyEJNS3_40alignment_score_matrix_one_column_bandedIiEENS3_34alignment_trace_matrix_full_bandedINS3_16trace_directionsELb0EEEEEENS1G_INS3_17affine_gap_policyEJiSt17integral_constantIbLb0EEEEENS1G_INS3_19find_optimum_policyEJEEENS1G_INS3_22affine_gap_init_policyEJEEENS1G_INS3_21scoring_scheme_policyEJS15_EEEEEEE9_M_invokeERKSt9_Any_dataOSL_OSX_ (std_function.h:290)
==47==    by 0x16E48F: operator() (std_function.h:590)
</details>

==47==    by 0x16E48F: execute<std::function<void(std::ranges::subrange<seqan3::detail::zip_view<std::ranges::single_view<std::tuple<std::vector<seqan3::dna5, std::allocator<seqan3::dna5> >&, std::vector<seqan3::dna5, std::allocator<seqan3::dna5> >&> >, std::ranges::iota_view<int, std::unreachable_sentinel_t> >::iterator<false>, seqan3::detail::zip_view<std::ranges::single_view<std::tuple<std::vector<seqan3::dna5, std::allocator<seqan3::dna5> >&, std::vector<seqan3::dna5, std::allocator<seqan3::dna5> >&> >, std::ranges::iota_view<int, std::unreachable_sentinel_t> >::iterator<false>, (std::ranges::subrange_kind)1>, std::function<void(seqan3::alignment_result<seqan3::detail::alignment_result_value_type<std::nullopt_t*, std::nullopt_t*, int, seqan3::detail::advanceable_alignment_coordinate<>, seqan3::detail::advanceable_alignment_coordinate<>, std::nullopt_t*, std::nullopt_t*, std::nullopt_t*> >)>)>&, std::ranges::subrange<seqan3::detail::zip_view<std::ranges::single_view<std::tuple<std::vector<seqan3::dna5, std::allocator<seqan3::dna5> >&, std::vector<seqan3::dna5, std::allocator<seqan3::dna5> >&> >, std::ranges::iota_view<int, std::unreachable_sentinel_t> >::iterator<false>, seqan3::detail::zip_view<std::ranges::single_view<std::tuple<std::vector<seqan3::dna5, std::allocator<seqan3::dna5> >&, std::vector<seqan3::dna5, std::allocator<seqan3::dna5> >&> >, std::ranges::iota_view<int, std::unreachable_sentinel_t> >::iterator<false>, (std::ranges::subrange_kind)1>, seqan3::detail::algorithm_executor_blocking<seqan3::detail::chunk_view<seqan3::detail::zip_view<std::ranges::single_view<std::tuple<std::vector<seqan3::dna5, std::allocator<seqan3::dna5> >&, std::vector<seqan3::dna5, std::allocator<seqan3::dna5> >&> >, std::ranges::iota_view<int, std::unreachable_sentinel_t> > >, std::function<void(std::ranges::subrange<seqan3::detail::zip_view<std::ranges::single_view<std::tuple<std::vector<seqan3::dna5, std::allocator<seqan3::dna5> >&, std::vector<seqan3::dna5, std::allocator<seqan3::dna5> >&> >, std::ranges::iota_view<int, std::unreachable_sentinel_t> >::iterator<false>, seqan3::detail::zip_view<std::ranges::single_view<std::tuple<std::vector<seqan3::dna5, std::allocator<seqan3::dna5> >&, std::vector<seqan3::dna5, std::allocator<seqan3::dna5> >&> >, std::ranges::iota_view<int, std::unreachable_sentinel_t> >::iterator<false>, (std::ranges::subrange_kind)1>, std::function<void(seqan3::alignment_result<seqan3::detail::alignment_result_value_type<std::nullopt_t*, std::nullopt_t*, int, seqan3::detail::advanceable_alignment_coordinate<>, seqan3::detail::advanceable_alignment_coordinate<>, std::nullopt_t*, std::nullopt_t*, std::nullopt_t*> >)>)>, seqan3::alignment_result<seqan3::detail::alignment_result_value_type<std::nullopt_t*, std::nullopt_t*, int, seqan3::detail::advanceable_alignment_coordinate<>, seqan3::detail::advanceable_alignment_coordinate<>, std::nullopt_t*, std::nullopt_t*, std::nullopt_t*> >, seqan3::detail::execution_handler_sequential>::fill_buffer()::<lambda(auto:78&&)> > (execution_handler_sequential.hpp:49)

https://github.com/seqan/seqan3/blob/fe7458e83b1415db38ff8148597017c3fabd030a/include/seqan3/core/algorithm/detail/execution_handler_sequential.hpp#L45-L50

==47==    by 0x16E48F: fill_buffer (algorithm_executor_blocking.hpp:262)

https://github.com/seqan/seqan3/blob/fe7458e83b1415db38ff8148597017c3fabd030a/include/seqan3/core/algorithm/detail/algorithm_executor_blocking.hpp#L248-L279

==47==    by 0x16E48F: next_result (algorithm_executor_blocking.hpp:199)

https://github.com/seqan/seqan3/blob/fe7458e83b1415db38ff8148597017c3fabd030a/include/seqan3/core/algorithm/detail/algorithm_executor_blocking.hpp#L192-L212

==47==    by 0x16E48F: next (algorithm_result_generator_range.hpp:139)

https://github.com/seqan/seqan3/blob/fe7458e83b1415db38ff8148597017c3fabd030a/include/seqan3/core/algorithm/algorithm_result_generator_range.hpp#L134-L146

==47==    by 0x16E48F: operator++ (algorithm_result_generator_range.hpp:234)

https://github.com/seqan/seqan3/blob/fe7458e83b1415db38ff8148597017c3fabd030a/include/seqan3/core/algorithm/algorithm_result_generator_range.hpp#L230-L236

==47==    by 0x16E48F: algorithm_range_iterator (algorithm_result_generator_range.hpp:202)

https://github.com/seqan/seqan3/blob/fe7458e83b1415db38ff8148597017c3fabd030a/include/seqan3/core/algorithm/algorithm_result_generator_range.hpp#L199-L203

==47==    by 0x16E48F: begin (algorithm_result_generator_range.hpp:102)

https://github.com/seqan/seqan3/blob/fe7458e83b1415db38ff8148597017c3fabd030a/include/seqan3/core/algorithm/algorithm_result_generator_range.hpp#L100-L103

==47==    by 0x16E48F: Alignment::bandedAlignment(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int) (pairwiseAlignment.cpp:105)

**Comment by marehr** [comment]
Okay, this definitely looks like some internal issue. (I annotated all the referenced code locations in your reply)

It would be awesome if you could provide `pairwiseAlignment.cpp:105` and your used sequences, if possible reduced, but you could also send them privately via https://gitter.im/seqan/Lobby or e-mail. The best form would be just the alignment code + all the sequences in a single cpp file. If it is huge then so be it, you can compress it :). If it is standalone, we can reduce it to a manageable size.
Are the sources available?

Some notes:
- alignment has been computed, and it aborts when constructing the final aligned sequence
- `begin (algorithm_result_generator_range.hpp:102)` suggests that it is the first alignment computation that fails. 
  ```c++
  auto range = align_pairwise(...);
  range.begin(); // computes first alignment
  ``` 
- https://github.com/seqan/seqan3/blob/fe7458e83b1415db38ff8148597017c3fabd030a/include/seqan3/alignment/matrix/detail/aligned_sequence_builder.hpp#L232-L244
  `(aligned_sequence_builder.hpp:243)` suggests that the trace-back somehow thinks that the (gapped) sequence is longer than the aligned sequence (`gap_decorator`) allows. This might be an off-by-one error, or worse a completely wrong span size. ~~It seems that it is in the "match" phase, as `n` added gaps would allow the iterator to go `n` places further~~ (this only holds true, if both sequences hold their invariants, but the traceback could mess that assumption up).
